src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -14310,6 +14874,385 @@ public abstract class MulticastDelegate : Delegate { }
                 Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "2m").WithArguments("System.Decimal").WithLocation(2, 24));
         }
 
+        [Theory, WorkItem(65728, "https://github.com/dotnet/roslyn/issues/65728")]
+        [InlineData("decimal ")]
+        [InlineData("decimal?")]
+        public void MissingDecimalConstantAttribute_Lambda(string type)
+        {
+            var source = $$"""
+                var lam = ({{type}} d = 1.1m) => { };
+                var lam2 = lam;
+                """;
+            var comp = CreateCompilation(source);
+            comp.MakeTypeMissing(WellKnownType.System_Runtime_CompilerServices_DecimalConstantAttribute);
+            comp.VerifyDiagnostics(
+                // (1,25): error CS0656: Missing compiler required member 'System.Runtime.CompilerServices.DecimalConstantAttribute..ctor'
```

Comment:
>// (1,25): error CS0656: Missing compiler required member 'System.Runtime.CompilerServices.DecimalConstantAttribute..ctor' [](http://example.com/codeflow?start=16&length=122)

I am curious what prevents us from reporting this error twice for this scenario. One for the lambda and one for the synthesized delegate.

Reply:
Good question, this is because the first error reported by `SourceComplexParameterSymbol` prevents the conversion from happening, i.e., `CheckLambdaConversion` isn't called. Specifically, `IsAnonymousFunctionCompatibleWithDelegate` checks for diagnostics from `UnboundLambda.Bind` (which creates the parameter symbol which adds the error) and `PreventsSuccessfulDelegateConversion` determines they prevent conversion.

Seems that `PreventsSuccessfulDelegateConversion` could be updated to ignore the missing type errors, because they don't prevent the conversion?

Reply:
> Thinking about it again, it seems that ```PreventsSuccessfulDelegateConversion``` could be updated to ignore the missing type errors, because they don't prevent the conversion?

Yes. I think we should do something about this. The errors that we report about synthesized attributes, default values, explicit attributes, params, etc. shouldn't have this effect on the binding. Let's cover synthesized attributes in this PR and open a follow-up issue to investigate if any other declaration related errors need special treatment as well.

Reply:
I have added missing type errors to the list of ignored errors. Also added tests (`*_OverloadResolution`). And created a follow up issue: https://github.com/dotnet/roslyn/issues/66060.